src/player.js

```diff
@@ -2003,6 +2002,9 @@ export default class Player extends FakeEventTarget {
    * @returns {void}
    */
   _play(): void {
+    if (!this._engine.src) {
+      this.load();
```

Comment:
This was removed in https://github.com/kaltura/playkit-js/pull/431, so need to find a solution that will accommodate both issues.
It is not possible to return this here due to other issues it creates.

Reply:
Why not? we called the internal load on the previous fix. 
this._load and then it wasn't pass through the load middleware. 
I called the load which pass through the load middleware.

Reply:
Please see related internal ticket for details.
If you are reverting a fix that was done just a few days before you need to check why it was done.

Reply:
maybe I miss something but I saw the ticket and the issue was when calling play() we called _load internally and it wasn't used in middleware

Reply:
@Yuvalke Your change makes the load middleware run after the play middleware finished. doesn't make sense.

Reply:
@yairans It's not after, as part of play middleware I use load middleware for specific use case.

Reply:
@Yuvalke `_play` is the last link in the `play` middleware chain.
Consider bumper for example - the `play`  middleware will run before the `load`. this is unacceptable.

Reply:
can't you just call to `_load` in attach flow?